src/index/art_index.cpp

```diff
@@ -89,12 +92,18 @@ bool ArtIndex::DeleteEntry(const storage::Tuple *key, ItemPointer *value) {
 
   if (removed) {
     // Update stats
-    DecreaseNumberOfTuplesBy(1);
-    if (static_cast<StatsType>(settings::SettingsManager::GetInt(
-            settings::SettingId::stats_mode)) != StatsType::INVALID) {
-      stats::BackendStatsContext::GetInstance()->IncrementIndexDeletes(
-          1, GetMetadata());
-    }
+    auto &stats_collector =
+        stats::ThreadLevelStatsCollector::GetCollectorForThread();
+    stats_collector.CollectIndexInsert(metadata->GetDatabaseOid(), GetOid());
+
+    // TODO: The memory collection here is just an inaccurate estimation
+    // Those who is familiar with the code base for index implementation should
+    // insert these lines to accurate place with correct values
+    size_t memory = key->GetLength() + 8;  // key size + item pointer size
+    stats_collector.CollectIndexMemoryAlloc(metadata->GetDatabaseOid(),
```

Comment:
It should be CollectIndexMemoryFree / Reclaim ??

-- Priyatham, Siva and Vamshi